docs/parse_argparse.py

```diff
@@ -0,0 +1,118 @@
+
+        path_temp_write_rst = os.path.abspath(path_temp_write)
+        #print(path_temp_write_rst)
+        base_path_temp_write = os.path.splitext(os.path.abspath(path_temp_write))[0]
+        #print(base_path_temp_write)
+        path_temp_write_rst = base_path_temp_write + '.rst'
+        #print(path_temp_write_rst)
+
+        #this block writes the argparse output to the respective '.rst' file with proper indentation
+        #this is the output that will be under '..code-block::' and will be displayed in html
+        with open(path_temp_write_rst, 'w+') as stdout:
+            subprocess.call(cmd, stdout=stdout, shell=True) #capture the argparse output
+            stdout.seek(0) #takes the cursor to beginning. Copy and indent stdout.
+            #The indentation is done because the arparse text needs to be indented
+            #under the ..code-block:: section in the final rst file
+            wrapper = textwrap.TextWrapper(initial_indent='\t', subsequent_indent='\t', width = 700)
+            wrapped=""
+            with open(path,'w') as fh_rst:
+                fh_rst.write('='*len_base + '\n' + basename + '\n' + '='*len_base + '\n' + '\n'+ '.. code-block:: shell' + '\n \n')
+                for line in stdout:
+                    wrapped = wrapped + wrapper.fill(line) + '\n' #copying the indented text
+                fh_rst.write(wrapped) #writing the indented tex to the final rst files, this will be dislayed in html
+        
+        #print(main_path)
+        main_file_no_rst = main_path + '_main' #making a file external_main
+        main_file = main_file_no_rst + '.rst' #making file external_main.rst
+        #print(main_file)
+        main_file = os.path.join(path_ori, main_file) # we now have /home/siddharth/Autometa/docs/source/scripts_2/common/external/external_main.rst
+        #print(main_file)
+        #print(file_name_base)
+
+        #this block will make the file 'script_main.rst' this is the script that will be called by the toctree in index.rst
+        if not os.path.exists('../docs/source/scripts_2/script_main.rst'):
+            with open ('../docs/source/scripts_2/script_main.rst', 'w') as fh_main:
+                fh_main.write('='*len('script_main') + '\n' + 'script_main \n' + '='*len('script_main') + '\n' + '\n'+ '.. toctree::' +
+                '\n \t :maxdepth: 3' + '\n \t :caption: Table of Contents \n')
+
+        #this will create a _main file for each directory. These rst files will be called by the toctreee in scripts_main.rst
+        if not os.path.exists(main_file):
+            len_head = len(main_path)
+            with open (main_file, 'w+') as fh_main:
+                fh_main.write('='*len_head + '\n' + main_path + '\n' + '='*len_head + '\n' + '\n'+ '.. toctree::' +
+                '\n \t :maxdepth: 2' + '\n \t :caption: Table of Contents \n' )
+
+            #need to create a link to external directory which is within the common directory
+            if main_path == 'common':
```

Comment:
Does this variable ever execute? In this case `main_path` is only consisting of `'common'`, how are you accounting for the path other than the name here?